tests/Unit/KinesisClientTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace PodPoint\KinesisLogger\Tests\Unit;
+
+use Mockery;
+use Illuminate\Support\Arr;
+use Aws\Kinesis\KinesisClient;
+use PodPoint\KinesisLogger\Tests\TestCase;
+
+class KinesisClientTest extends TestCase
+{
+    /**
+     * @return array
+     */
+    public function loggerLevelTestProvider(): array
+    {
+        return [
+            ['debug'],
+            ['info'],
+            ['notice'],
+            ['warning'],
+            ['error'],
+            ['critical'],
+            ['alert'],
+        ];
+    }
+
+    /**
+     * @dataProvider loggerLevelTestProvider $logLevel
+     */
+    public function testLoggerLevels($logLevel)
+    {
+        $this->app['config']->set('logging.channels', [
+            'kinesis' => [
+                'driver' => 'kinesis',
+                'stream' => 'logging',
+                'level' => 'debug',
+            ],
+        ]);
+
+        $this->app['config']->set('logging.default', 'kinesis');
+
+        $mocked = $this->getMockedKinesisClient();
+
+        $mocked->shouldReceive('putRecord')->once()->with(Mockery::on(function ($argument) use ($logLevel) {
+            $data = json_decode($argument['Data'], true);
+            return $data['level'] == strtoupper($logLevel);
+        }));
+
+        $this->app->instance(KinesisClient::class, $mocked);
+
+        logger()->$logLevel("Test {$logLevel} message");
+    }
+
+    public function testLoggerDoesNotLogBelowDefaultInfoLevel()
+    {
+        $this->app['config']->set('logging.channels', [
+            'kinesis' => [
+                'driver' => 'kinesis',
+                'stream' => 'myStream',
+            ],
+        ]);
+
+        $this->app['config']->set('logging.default', 'kinesis');
+
+        $mocked = $this->getMockedKinesisClient();
+
+        $mocked->shouldNotReceive('putRecord');
+
+        $this->app->instance(KinesisClient::class, $mocked);
+
+        logger()->debug('Test debug message');
+    }
+
+    public function testLoggerLogsAboveDefaultInfoLevel()
+    {
+        $this->app['config']->set('logging.channels', [
+            'kinesis' => [
+                'driver' => 'kinesis',
+                'stream' => 'myStream',
+            ],
+        ]);
+
+        $this->app['config']->set('logging.default', 'kinesis');
+
+        $mocked = $this->getMockedKinesisClient();
+
+        $mocked->shouldReceive('putRecord')->twice();
+
+        $this->app->instance(KinesisClient::class, $mocked);
+
+        logger()->info('Test info message');
+        logger()->warning('Test warning message');
+    }
+
+    /**
+     * @dataProvider loggerLevelTestProvider $logLevel
+     */
+    public function testLoggerDoesNotLogBelowMinimumLevel($logLevel)
```

Comment:
I think the dataprovider is overkill here - we could probably just use the same methodology as above and use `twice` on the mock expectation and send two below, and one at the same level